File: app/src/main/java/com/erebor/tomkins/pos/di/LogModule.java
package com.erebor.tomkins.pos.di;

import android.util.Log;

import com.erebor.tomkins.pos.tools.Logger;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class LogModule {

    @Provides
    @Singleton
    Logger provideLogger() {
        return new Logger() {
            @Override
            public void info(String tag, String message) {
                Log.i(tag, message);
            }

            @Override
            public void debug(String tag, String message) {
                Log.d(tag, message);
            }

            @Override
            public void warning(String tag, String message) {
                Log.w(tag, message);
            }

            @Override
            public void error(String tag, String message, Throwable throwable) {
                Log.e(tag, message, throwable);

            }
        };
    }

}

File: app/src/main/java/com/erebor/tomkins/pos/di/LogModule.kt
package com.erebor.tomkins.pos.di

import android.util.Log

import com.erebor.tomkins.pos.tools.Logger

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
class LogModule {

    @Provides
    @Singleton
    internal fun provideLogger(): Logger {
        return object : Logger {
            override fun info(tag: String, message: String) {
                Log.i(tag, message)
            }

            override fun debug(tag: String, message: String) {
                Log.d(tag, message)
            }

            override fun warning(tag: String, message: String) {
                Log.w(tag, message)
            }

            override fun error(tag: String, message: String, throwable: Throwable) {
                Log.e(tag, message, throwable)

            }
        }
    }

}
